File: 241008/요일 맞추기/guess-day-of-week.cpp
#include <iostream>
using namespace std;

int days_in_month(int month) {
    int days_in_months[] = { 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
    return days_in_months[month - 1];
}

string get_weekday(int total_days) {
    string weekdays[] = { "Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun" };
    return weekdays[total_days % 7];
}

int main() {
    int m1, d1, m2, d2;
    cin >> m1 >> d1 >> m2 >> d2;

    int total_days = 0;

    if (m1 == m2) {
        total_days = d2 - d1;
    } else {
        total_days += days_in_month(m1) - d1;
        for (int i = m1 + 1; i < m2; i++) {
            total_days += days_in_month(i);
        }
        total_days += d2;
    }

    cout << get_weekday(total_days) << endl;

    return 0;
}

File: 241022/요일 맞추기/guess-day-of-week.py
m1, d1, m2, d2 = tuple(map(int, input().split()))
ans = 0 
ar = [31, 28 ,31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
week = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"]

while 1 :
    if m1 == m2 and d1 == d2 :
        break

    if m1 < m2 :
        d1 += 1
        ans += 1

    elif m1 > m2 :
        d1 -= 1
        ans += 1

    elif m1 == m2 and d1 > d2 :
        d1 -= 1
        ans + 1
    else :
        d1 += 1
        ans += 1
        
    if d1 > ar[m1-1] :
        m1 += 1
        d1 = 1

    elif d1 == 0 :
        m1 -= 1
        d1 = ar[m1-1]

print(week[ans//7])